src/test/java/com/microsoft/sqlserver/jdbc/resiliency/BasicConnectionTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc.resiliency;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Map;
+
+import javax.sql.PooledConnection;
+
+import org.junit.jupiter.api.Test;
+
+import com.microsoft.sqlserver.jdbc.RandomUtil;
+import com.microsoft.sqlserver.jdbc.SQLServerConnectionPoolDataSource;
+import com.microsoft.sqlserver.jdbc.TestResource;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+
+
+public class BasicConnectionTest extends AbstractTest {
+
+    @Test
+    public void testBasicReconnectDefault() throws SQLException {
+        basicReconnect(connectionString);
+    }
+
+    @Test
+    public void testBasicEncryptedConnection() throws SQLException {
+        basicReconnect(connectionString + ";encrypt=true;trustServerCertificate=true;");
+    }
+
+    @Test
+    public void testGracefulClose() throws SQLException {
+        try (Connection c = ResiliencyUtils.getConnection(connectionString)) {
+            try (Statement s = c.createStatement()) {
+                ResiliencyUtils.killConnection(c, connectionString);
+                c.close();
+                s.executeQuery("SELECT 1");
+                fail("Query execution did not throw an exception on a closed execution");
+            } catch (SQLException e) {
+                assertTrue(e.getMessage().contains("The connection is closed."));
+            }
+        }
+    }
+
+    @Test
+    public void testSetAttributes() throws SQLException {
+        try (Connection c = ResiliencyUtils.getConnection(connectionString)) {
+            ResiliencyUtils.toggleRandomProperties(c);

```

Comment:
I think we need to separate these based on server support. e.g. SET REMOTE_PROC_TRANSACTIONS is not supported in Azure DB

Reply:
update - this is deprecated, we can remove this as per David

Reply:
Removed.